### PR TITLE
DAOS-7093 test: Refactor container/attribute.py

### PR DESCRIPTION
**Pull request by shimizukko**
- Fix `DaosContainer.list_attr()` because it was broken.
- Fix `DaosContainer.get_attr()` to support asynchronous mode.
- Use API instead of daos command throughout the tests.
- Remove `test_container_large_attributes()` because this is covered in query_attribute.py
- Change the name to api_basic_attribute.py because this test covers the single name for set/list/get attribute. Multiple names will be covered in another test.
- Refactored the code.

Skip-unit-tests: true
Test-tag: container_api_basic_attribute_sync container_api_basic_attribute_async dm_preserve_props_fs_copy_posix_dfs dm_dst_create
Signed-off-by: Makito Kano <makito.kano@intel.com>

### CODE REVIEW

**Review by dinghwah — APPROVED** [review]
LGTM

File: src/tests/ftest/container/api_basic_attribute.py [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+    def verify_get_attr(self, indata, outdata):
+        """Verify the Attributes value after get_attr.
+
+        Args:
+            indata (dict): Name-value pairs that were used for set-attr.
+            outdata (dict): Name-value pairs that were obtained from get-attr.
+        """
+        decoded = {}
+        for key, val in outdata.items():
+            decoded[key.decode()] = val
```
Comment (daltonbohning) [inline src/tests/ftest/container/api_basic_attribute.py:87]: Just FYI, this also works
```suggestion
        decoded = {key.decode(): val for key, val in outdata.items()}
```

**Review by daltonbohning — APPROVED** [review]
LGTM

File: src/tests/ftest/container/api_basic_attribute.py (hunk 1 above)
Comment (shimizukko) [reply src/tests/ftest/container/api_basic_attribute.py:87]: Okay, thanks!

**Pull request merged**